gecko/core/container.py

```
@@ -273,22 +342,113 @@ def _create_instance(self, descriptor: ServiceDescriptor[T]) -> T:
+        # 解析依赖，支持 Optional[...] 等 typing 包装类型
         kwargs = {}
         for param_name, param_type in hints.items():
-            if param_type in self._services or (self._parent and param_type in self._parent._services):
-                kwargs[param_name] = self.resolve(param_type)
-        
+            origin = get_origin(param_type)
+            target_type = None
+            if origin is Union:
+                # 处理 Optional[T] => Union[T, NoneType]
+                args = get_args(param_type)
+                non_none = [a for a in args if a is not type(None)]  # noqa: E721
```

Ruff (0.14.7)

Unused `noqa` directive (unused: `E721`)

Remove unused `noqa` directive

(RUF100)